sn_client/src/files.rs

```diff
@@ -185,11 +194,37 @@ impl FilesApi {
     }
 }
 
-/// Encrypts a [`LargeFile`] and returns the resulting address and all chunk names.
-/// Correspondent encrypted chunks are written in the specified output folder.
-/// Does not store anything to the network.
-///
-/// Returns data map as a chunk, and the resulting chunks
+struct EmptyFile {}
+
+struct SmallFile {
+    buffer: Vec<u8>,
+}
+
+impl SmallFile {
+    fn new(buffer: Vec<u8>) -> Self {
+        Self { buffer }
+    }
+}
+
+trait ChunkBySize {
+    fn chunk_by_size(self) -> (Chunk, Vec<(XorName, PathBuf)>);
+}
```

Comment:
I'm not sure if we need extra structs and traits here? The logic is small enough that it can be moved inside `chunk_file()`?

Reply:
https://blog.logrocket.com/build-polymorphic-components-rust/#understanding-polymorphism-rust